rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,234 @@
+                }
+            }
+        else:
+            quarantine_payload = {
+                "query": "mutation( $orgID:String! $agentID:String!) { unquarantineAssets( orgId:$orgID assetIds: [$agentID] ) { results { assetId failed } } }",
+                "variables": {
+                    "orgID": self.org_key,
+                    "agentID": agent_id
+                }
+            }
+
+        action_verb = "quarantine" if quarantine_state else "unquarantine"
+        self.logger.info(f"Attempting to {action_verb} asset {agent_id} at {self.endpoint}")
+
+        results_object = self._post_payload(quarantine_payload)
+
+        if quarantine_state:
+            failed = results_object.get("data").get("quarantineAssets").get("results")[0].get("failed")
+        else:
+            failed = results_object.get("data").get("unquarantineAssets").get("results")[0].get("failed")
+
+        return (not failed)
+
+    def connection_test(self):
+        # Return the first org to verify the connection works
+        graph_ql_payload = {
+            "query":"{ organizations(first: 1) { edges { node { id } } totalCount } }"
+        }
+
+        # If no exceptions are thrown, we have a valid connection
+        self._post_payload(graph_ql_payload)
+        return True
+
+    #################
+    # Private Methods
+    #################
+
+    def _get_headers(self):
+        """
+        This build and returns headers for the request session
+
+        :return: dict
+        """
+        return {
+            "X-Api-key": self.api_key,
+            "Accept-Version": "kratos",
+            "Accept": "*/*",
+            "Accept-Encoding": "gzip, deflate, br"
+        }
+
+    def _post_payload(self, payload):
+        """
+        This will post a given payload to the API using the connection session
+
+        :param payload: dict
+        :return: dict
+        """
+        result = self.session.post(self.endpoint, json=payload)
+
+        try:
+            result.raise_for_status()
+        except:
+            raise APIException(cause="Error connecting to the Insight Agent API.",
+                               assistance="Please check your Org ID and API key.\n",
+                               data=result.text)
+
+        results_object = result.json()
+
+        if results_object.get("errors"):
+            raise APIException(cause="The Insight Agent API returned errors",
+                               assistance=results_object.get("errors"))
+
+        return results_object
+
+    def _setup_endpoint(self, region_string):
+        region_code = region_map.get(region_string)
+
+        if region_code:
+            endpoint = f"https://{region_code}.api.insight.rapid7.com/graphql"
+        else:
+            # It's an enum, hopefully this never happens.
+            raise APIException(cause="Region not found.",
+                               assistance="Region code was not found for selected region. Please contact support.")
```

Comment:
```suggestion
                               assistance="Region code was not found for selected region. Available regions are: ...")
```
Fill in the regions for that message - but a library that is potentially re-used somewhere outside a plugin (or maybe even in some third party code) shouldn't tell the developer/user to contact support :)